pkg/controllers/networkinfo/networkinfo_controller.go

```diff
@@ -185,31 +185,34 @@ func (r *NetworkInfoReconciler) Reconcile(ctx context.Context, req ctrl.Request)
 	retryWithSystemVPC := false
 	var systemNSCondition *corev1.NamespaceCondition
 
-	gatewayConnectionReady, _ := getGatewayConnectionStatus(ctx, systemVpcNetCfg)
-	gatewayConnectionReason := ""
-	if !gatewayConnectionReady {
+	gatewayConnectionReady, _ := getGatewayConnectionStatus(systemVpcNetCfg)
+	serviceClusterReady, _ := getServiceClusterStatus(systemVpcNetCfg)
+
+	if !gatewayConnectionReady && !serviceClusterReady {
 		// Retry after 60s if the gateway connection is not ready in system VPC.
 		if ncName != commonservice.SystemVPCNetworkConfigurationName {
-			log.Info("Skipping reconciliation due to unready system gateway connection", "NetworkInfo", req.NamespacedName)
-			setNSNetworkReadyCondition(ctx, r.Client, req.Namespace, nsMsgVPCGwConnectionNotReady.getNSNetworkCondition())
+			log.Info("Skipping reconciliation due to unready system gateway connection and service cluster", "NetworkInfo", req.NamespacedName)
+			setNSNetworkReadyCondition(ctx, r.Client, req.Namespace, nsMsgVPCConnectionNotReady.getNSNetworkCondition())
 			return common.ResultRequeueAfter60sec, nil
 		}
 
-		// Re-check the gateway connection readiness in system VPC on NSX.
-		gatewayConnectionReady, gatewayConnectionReason, err = r.Service.ValidateGatewayConnectionStatus(nc)
-		log.Info("Got the gateway connection status", "gatewayConnectionReady", gatewayConnectionReady, "gatewayConnectionReason", gatewayConnectionReason)
+		// Re-check the gateway connection and service cluster readiness in system VPC on NSX.
+		connectionStatus, err := r.Service.ValidateConnectionStatus(nc)
 		if err != nil {
+			log.Error(err, "Failed to get the connection status")
 			r.StatusUpdater.UpdateFail(ctx, networkInfoCR, err, fmt.Sprintf("Failed to validate the edge and gateway connection, Project: %s", nc.Spec.NSXProject), setNetworkInfoVPCStatusWithError, nil)
 			setNSNetworkReadyCondition(ctx, r.Client, req.Namespace, nsMsgVPCGwConnectionGetError.getNSNetworkCondition(err))
 			return common.ResultRequeueAfter10sec, err
 		}
-		setVPCNetworkConfigurationStatusWithGatewayConnection(ctx, r.Client, systemVpcNetCfg, gatewayConnectionReady, gatewayConnectionReason)
+		log.Info("Got the connection status", "status", connectionStatus)
+		setVPCNetworkConfigurationStatusWithGatewayConnection(ctx, r.Client, systemVpcNetCfg, connectionStatus)
+		serviceClusterReady = connectionStatus.GatewayConnectionReady
```

Comment:
Should `serviceClusterReady` be ServiceClusterReady?

Reply:
You're right, updated, thanks!